ggshield/verticals/secret/secret_scan_collection.py

```diff
@@ -12,24 +13,41 @@
 from ggshield.verticals.secret.extended_match import ExtendedMatch
 
 
-class Result(NamedTuple):
+class Result:
     """
     Return model for a scan which zips the information
     between the Scan result and its input file.
     """
 
-    # TODO: Rename `file` to `scannable`?
-    file: Scannable  # filename that was scanned
+    filename: str  # Name of the file/patch scanned
+    filemode: Filemode
+    filepath: Path
+    file_url: str
```

Comment:
Since not all scannables are files (cf the TODO comment), I propose the following name changes:

- filepath -> path
- file_url -> url